src/kyoto_reader/reader.py

```diff
@@ -110,6 +177,21 @@ def read_knp(path: Path, did_from_sid: bool) -> Dict[str, str]:
                 did2knps[did] = buff
             else:
                 logger.warning(f'empty file found and skipped: {path}')
+
+        if archive is not None:
+            with archive.open(str(path)) as f:
+                text = f.read().decode("utf-8")
+                _read_knp(text.split("\n"))
+        else:
+            if any(key in path.suffixes for key in KyotoReader.COMPRESS2OPENER):
+                compress = set(path.suffixes) & set(KyotoReader.COMPRESS2OPENER.keys())
+                assert len(compress) == 1
+                opener = KyotoReader.COMPRESS2OPENER[compress.pop()]
+            else:
+                opener = open
```

Comment:
A bit strange substitution? ("opener" is a kind of actor, and "open" is a kind of action)